origin-dapp/public/swAnalytics.js

```diff
@@ -0,0 +1,89 @@
+// Based off
+// https://developers.google.com/web/ilt/pwa/integrating-analytics
+
+class Analytics {
+    constructor(){
+        this.trackingId = undefined
+    }
+    setTrackingId(trackingId){
+        this.trackingId = trackingId
+    }
+    event(eventCategory, eventAction, eventLabel) {
+        'use strict';
+        
+        // console.log('Sending analytics event: ' + eventCategory + '/' + eventAction);
+        const trackingId = this.trackingId
+        if (!trackingId) {
+            console.error('GA Tracking Id not set in serviceworker analytics-helper.js');
+            // We want this to be a safe method, so avoid throwing unless absolutely necessary.
+            return Promise.resolve();
+        }
+        
+        if (!eventAction && !eventCategory) {
+            console.warn('sendAnalyticsEvent() called with no eventAction or ' +
+            'eventCategory.');
+            // We want this to be a safe method, so avoid throwing unless absolutely necessary.
+            return Promise.resolve();
+        }
+        
+        return self.registration.pushManager.getSubscription()
+        .then(function(subscription) {
+            if (subscription === null) {
+            throw new Error('No subscription currently available.');
+            }
+        
+            // Create hit data
+            var payloadData = {
+            // Version Number
+            v: 1,
+            // Client ID
+            cid: subscription.endpoint,
+            // Tracking ID
+            tid: trackingId,
+            // Hit Type
+            t: 'event',
+            // Event Category
+            ec: eventCategory,
+            // Event Action
+            ea: eventAction,
+            // Event Label
+            el: eventLabel
+            };
+        
+            // Format hit data into URI
+            var payloadString = Object.keys(payloadData)
```

Comment:
@DanielVF any reason not to use `const` here? I did in the `sw.js`, which I doubt gets transpiled.

Reply:
We can use it.